features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListPresenter.kt

```diff
@@ -115,19 +116,47 @@ class DefaultActionListPresenter @AssistedInject constructor(
             isEventPinned = pinnedEventIds.contains(timelineItem.eventId),
         )
 
-        val displayEmojiReactions = usersEventPermissions.canSendReaction &&
-            timelineItem.content.canReact()
-        if (actions.isNotEmpty() || displayEmojiReactions) {
+        val verifiedUserSendFailure = buildVerifiedUserSendFailure(timelineItem)
+        val displayEmojiReactions = usersEventPermissions.canSendReaction && timelineItem.content.canReact()
+
+        if (actions.isNotEmpty() || displayEmojiReactions || verifiedUserSendFailure != ActionListState.VerifiedUserSendFailure.None) {
             target.value = ActionListState.Target.Success(
                 event = timelineItem,
                 displayEmojiReactions = displayEmojiReactions,
+                verifiedUserSendFailure = verifiedUserSendFailure,
                 actions = actions.toImmutableList()
             )
         } else {
             target.value = ActionListState.Target.None
         }
     }
 
+    private suspend fun buildVerifiedUserSendFailure(
+        timelineItem: TimelineItem.Event,
+    ): ActionListState.VerifiedUserSendFailure {
+        return when (val sendState = timelineItem.localSendState) {
+            is LocalEventSendState.Failed.VerifiedUserHasUnsignedDevice -> {
+                val userId = sendState.devices.keys.firstOrNull()
+                if (userId == null) {
+                    ActionListState.VerifiedUserSendFailure.None
```

Comment:
Is this OK? I guess it shouldn't ever happen except if there is some bug in the SDK, but maybe we should still display the UI with some placeholder value and log the issue?